### PR TITLE
PROJECT: TIC TAC TOE

### PR DESCRIPTION
**Pull request by macnick**
We have made a simple Tic Tac Toe game you that you can play in your browser!

The users can
1. . Fill in their names in the boxes
2. Click on Play to start a new game
3. If there is a winner or the game finishes with a tie click on the popup message to make it go away
4. Click on Play to play again

We created separate modules for Board and Player. Also the whole system is an MVC design. The logic is in the controller module.  The view deals only with  screen drawing. 

Please review our code

### CODE REVIEW

File: assets/index.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
+/* eslint-disable no-use-before-define */
+/* eslint-disable no-unused-expressions */
+/* eslint-disable no-unused-vars */
+/* eslint-disable import/extensions */
+import Player from './player.js';
+import Table from './board.js';
+
```
Comment (bellom) [inline assets/index.js]: clear code!
kindly move the DOM functions to another file.
image

File: assets/index.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        const msg = 'It is a tie!';
+        view.showModal(msg);
+      }
+      // change player
+      currentPlayer = currentPlayer === p1 ? p2 : p1;
+      highlightPlayer();
+    }
+  };
+
+  const isItATie = () => t.t.every((pos) => pos != null);
```
Comment (bellom) [inline assets/index.js]: [Optional]
Am not good in naming too 😄 but you can name this `checkTie` right?
seems your function name is a complete sentence 🙂

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**_1. Kindly check out my comments below;_**

**_2. Kindly make typing of user's name required_**

**_3. Kindly add `reset` button;_**
image


**Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....**

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

File: assets/index.js (hunk 2 above)
Comment (macnick) [reply assets/index.js]: Done!

File: assets/index.js (hunk 1 above)
Comment (macnick) [reply assets/index.js]: Done!

**Comment by macnick** [comment]
Thank you @bellom for your kind and prompt review. 👍 
We fixed the 2 things you suggested. 
1. Moved the DOM module to a separate file.
2. Renamed the isItATie to checkTie which makes more sense. 

Regarding the requirements 2 and 3 we have some objection. Please read and think about them.  

According to instructions :
- "Clean up the interface to **allow** players to put in their names,"
We have to **allow** the player to add a name, not to **enforce** the name input. As it is, the game allows the player to do whatever he likes. More freedom. 
- " include a button to start/restart the game" We already have a button to start/restart the game. We do not need a second button.
- "and add a display element that congratulates the winning player!"
After a player wins he/she gets a message "Winner is PlayerName". Do we really have to change it to read: "Congratulations PlayerName you won!" ? 

Thank you for your patience. 

Please review our code and play the game with a friend. 🙏

**Comment by bellom** [comment]
Hey @macnick, 

Thanks for the professional reply.
2 and 3 should have been tag as optional 🙏 

Let's move to next project guys 👍 

Have you guys submitted another `CRR`?

**Comment by addod19** [comment]
> Hey @macnick,
> 
> Thanks for the professional reply.
> 2 and 3 should have been tag as optional 🙏
> 
> Let's move to next project guys 👍
> 
> Have you guys submitted another `CRR`?

Yes we have.

**Review by Oghenebrume50 — APPROVED** [review]
### STATUS: APPROVED 

Hi, Team this is awesome, all corrections from the previous review were made project looks good now, please merge and move on :+1: **This is approved** :slightly_smiling_face: 

**RAPHAEL NORIODE** :sunglasses:

**Comment by addod19** [comment]
> ### STATUS: APPROVED
> Hi, Team this is awesome, all corrections from the previous review were made project looks good now, please merge and move on 👍 **This is approved** 🙂
> 
> **RAPHAEL NORIODE** 😎

Thank you for the quick review 👐

**Pull request merged**